github/resource_github_repository_test.go

```diff
@@ -693,11 +693,11 @@ func testAccCreateRepositoryBranch(branch, repository string) error {
 		Owner:   testOwner,
```

Comment:
hmm looks like it'll be problematic anywhere `Owner:   testOwner` is used in tests ...might make sense to determine the value in provider_utils.go from either of the 2 env vars

Reply:
actually thinking about this more, since this function is only referenced in one PreConfig @L286, it's not too drastic to update this function to use the provider so lines 687-699 can be eliminated and line 700 can be updated to use the provider meta e.g.
```
client := testAccProvider.Meta().(*Owner).v3client
```

and then the Preconfig ref can be simplified to: 
```				
PreConfig: func() { testAccCreateRepositoryBranch("foo", *repo.Name) },
```

Reply:
In the short term as we are building out GHES acceptance testing, my plan is to perform passes over our existing test cases to modernize / minimize where needed.  I'm not fully understanding the change here, but welcome a commit either on this PR or another one that follows.  However in general, I'm hoping that once [tests are expanded](https://github.com/terraform-providers/terraform-provider-github/pull/523), we can align our tests to best fit an Action-oriented test suite.